scheduler/reconcile.go

```diff
@@ -416,6 +417,11 @@ func (a *allocReconciler) computeLimit(group *structs.TaskGroup, untainted, dest
 	if a.deployment != nil {
 		partOf, _ := untainted.filterByDeployment(a.deployment.ID)
 		for _, alloc := range partOf {
+			// An unhealthy allocation means nothing else should be happen.
+			if alloc.DeploymentStatus.IsUnhealthy() {
+				return 0
+			}
+
 			if !alloc.DeploymentStatus.IsHealthy() {
 				limit--
```

Comment:
It is still reachable. If the allocation hasn't set either healthy or unhealthy!